### PR TITLE
0.3.0

### PR DESCRIPTION
**Pull request by sjud**
Hi Sean,

I hope this message finds you well. First off, I'd like to express my gratitude for your encouragement in the leptos Discord. It's been a great source of motivation.

Context of This Pull Request:
This pull request is somewhat unconventional in its nature, stemming from work I embarked on prior to our connection. Consequently, it doesn't adhere strictly to the typical pull request etiquette or social protocol. It's comprehensive, encompassing around 800 lines of changes, including API modifications, reorganization of code, and deletion of some comments.

Key Changes

API and Structural Adjustments: I've made several changes to the API and the overall structure of the code. This includes moving elements around and removing certain parts. Notably, I've eliminated some comments where I believed the modifications or the new Axum API rendered them obsolete.
HttpBody Implementation: Specific alterations have been made to the HttpBody implementation, aligning it with the latest API standards.
Axum Body::channel Replacement: Given the removal of axum::Body::channel, I've opted for the channel provided in the async-channel crate, which was already included in our dependencies.

Additions and Renovations

Introduced a GitHub workflow.
Added a work-in-progress leptos example.
Established a tests folder, migrating the test module into this directory.
Renamed server to oneshot to better reflect its functionality and to accommodate a genuine server function capable of processing a stream of web_sys requests.
Updated the Http body to comply with the latest API.
Reformed function bodies to utilize the new API.
Upgraded all crates to their latest versions.
Updated the crate version to 0.3.0 to account for breaking changes.

Looking Forward:
I'm eager to collaborate with you on refining and enhancing this project. Please know that I'm highly receptive to feedback, including critiques and suggestions. I view this endeavor as a valuable learning opportunity and am keen on improving my contributions. Your innovative approach to developing this crate has been both inspiring and educational for me, and I'm grateful for the insights I've gained from your work.

Conclusion:
I believe these changes, while extensive, are a step toward enhancing the project's functionality and adaptability. I look forward to your thoughts and any feedback you may have.

Best regards,
Sam

### CODE REVIEW

File: examples/leptos_counter_isomorphic/.DS_Store
Comment (seanaye) [inline examples/leptos_counter_isomorphic/.DS_Store:1]: Remove DS_Store

File: src/lib.rs [hunk 1]
```diff
@@ -54,13 +48,20 @@ where
...
+                    Ok(b) => {
+                        let slice = b.chunk();
+                        let a = Uint8Array::new(&JsValue::from(slice.len()));
+                        a.copy_from(slice);
+                        let out: JsValue = a.into();
+                        Some(Ok(out))
+                    }
+                    // Not 100% sure how to best handle this error besides informing the user.
+                    // This gives a frame that isn't a data frame and I don't know how that could be useful here?
+                    Err(_) => Some(Err(JsValue::from_str("Not a data frame"))),
```
Comment (seanaye) [inline src/lib.rs:63]: Good comment, I think this behaviour is fine for now

File: tests/tests.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        for i in 0..10 {
+            // post i to the server under url count/{i}
+        }
+        // close server
+    });
+    let mut i = 0;
+    while Ok(msg) = rx.recv() {
+        assert_eq!(JsFuture::form(msg).await.unwrap(),i+1);
+    }
+}*/
```
Comment (seanaye) [inline tests/tests.rs:93]: remove comment if it isn't implemented yet

File: src/lib.rs [hunk 3]
```diff
@@ -178,54 +165,28 @@ impl App {
...
+
+    // TODO
+    /*
+    pub async fn serve(&self, ?) -> ? {
+        Listen for a stream of web_sys::Requests and return a stream of web_sys::Responses
+        recreating the server functionality in a wasm environment.
     }
+    */
 }
+
```
Comment (seanaye) [inline src/lib.rs:192]: Remove if not yet implemented

**Review by seanaye — COMMENTED** [review]
Excellent work, thanks for your contribution. Could we just clean up the comments of not-yet-implemented things, remove DS_Store, etc

**Comment by sjud** [comment]
Awesome, thank you!

File: src/lib.rs (hunk 1 above)
Comment (sjud) [reply src/lib.rs:62]: Thanks!

File: examples/leptos_counter_isomorphic/.DS_Store
Comment (sjud) [reply examples/leptos_counter_isomorphic/.DS_Store:1]: Done

File: src/lib.rs (hunk 3 above)
Comment (sjud) [reply src/lib.rs]: Done

File: tests/tests.rs (hunk 2 above)
Comment (sjud) [reply tests/tests.rs]: Done

**Pull request merged**